sympy/core/tests/test_relational.py

```diff
@@ -599,15 +601,66 @@ def test_simplify_relational():
     # _eval_simplify routine
     assert simplify(-(2**(3*pi/2) + 6**pi)**(1/pi) +
         2*(2**(pi/2) + 3**pi)**(1/pi) < 0) is S.false
+
     # canonical at least
-    for f in (Eq, Ne):
-        f(y, x).simplify() == f(x, y)
-        f(x - 1, 0).simplify() == f(x, 1)
-        f(x - 1, x).simplify() == S.false
-        f(2*x - 1, x).simplify() == f(x, 1)
-        f(2*x, 4).simplify() == f(x, 2)
-        z = cos(1)**2 + sin(1)**2 - 1  # z.is_zero is None
-        f(z*x, 0).simplify() == f(z*x, 0)
+    assert Eq(y, x).simplify() == Eq(x, y)
+    assert Eq(x - 1, 0).simplify() == Eq(x, 1)
+    assert Eq(x - 1, x).simplify() == S.false
+    assert Eq(2*x - 1, x).simplify() == Eq(x, 1)
+    assert Eq(2*x, 4).simplify() == Eq(x, 2)
+    z = cos(1)**2 + sin(1)**2 - 1  # z.is_zero is None
+    assert Eq(z*x, 0).simplify() == S.true
```

Comment:
I'm not sure this should happen. How did it know that z was 0?

Reply:
Z is defined on the above line and is simplified to 0.

(Note that earlier there were no assert for these "tests"...)